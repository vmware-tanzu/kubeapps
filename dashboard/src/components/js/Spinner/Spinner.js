// Copyright 2020-2022 the Kubeapps contributors.
// SPDX-License-Identifier: Apache-2.0

import cs from "classnames";
import PropTypes from "prop-types";
import React from "react";
import "./Spinner.scss";

/**
 * https://v2.clarity.design/spinners
 */
const Spinner = ({ center, inline, inverse, medium, small, text }) => {
  const cssClass = cs("spinner", {
    "spinner-center": center && !inline,
    "spinner-inline": inline,
    "spinner-inverse": inverse,
    "spinner-md": medium,
    "spinner-sm": small,
  });

  const containerCssClass = cs("spinner-container", {
    "spinner-center": center && !inline,
  });

  return (
    <span className={containerCssClass}>
      <span className={cssClass}>{text}</span>
    </span>
  );
};

Spinner.propTypes = {
  center: PropTypes.bool,
  inline: PropTypes.bool,
  inverse: PropTypes.bool,
  medium: PropTypes.bool,
  small: PropTypes.bool,
  text: PropTypes.string,
};

Spinner.defaultProps = {
  center: true,
  inline: false,
  inverse: false,
  medium: false,
  small: false,
  text: "Loading...",
};

export default Spinner;
