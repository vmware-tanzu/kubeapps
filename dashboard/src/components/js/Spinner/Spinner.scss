// Copyright 2020-2022 the Kubeapps contributors.
// SPDX-License-Identifier: Apache-2.0

.spinner {
  &-container {
    &.spinner-center {
      display: block;
      text-align: center;
    }
  }

  &.spinner-inline {
    margin-right: 6px;
  }
}
