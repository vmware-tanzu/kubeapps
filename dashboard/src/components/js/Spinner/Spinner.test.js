// Copyright 2020-2022 the Kubeapps contributors.
// SPDX-License-Identifier: Apache-2.0

import { shallow } from "enzyme";
import React from "react";
import Spinner from ".";

describe(Spinner, () => {
  it("displays the basic spinner", () => {
    const wrapper = shallow(<Spinner />);
    expect(wrapper).toMatchSnapshot();
  });

  it("sets a different text", () => {
    const text = "My Text";
    const wrapper = shallow(<Spinner text={text} />);

    expect(wrapper).toHaveText(text);
  });

  it("applies the center style based on props", () => {
    const wrapper = shallow(<Spinner center />);

    expect(wrapper).toHaveClassName("spinner-center");
  });

  it("applies the inline style based on props", () => {
    const wrapper = shallow(<Spinner inline />);

    expect(wrapper.children().first()).toHaveClassName("spinner-inline");
  });

  it("ignores the center style if the inline mode is enabled", () => {
    const wrapper = shallow(<Spinner center inline />);

    expect(wrapper).not.toHaveClassName("spinner-center");
  });

  it("applies the inverse style based on props", () => {
    const wrapper = shallow(<Spinner inverse />);

    expect(wrapper.children().first()).toHaveClassName("spinner-inverse");
  });

  it("applies the small style based on props", () => {
    const wrapper = shallow(<Spinner small />);

    expect(wrapper.children().first()).toHaveClassName("spinner-sm");
  });

  it("applies the medium style based on props", () => {
    const wrapper = shallow(<Spinner medium />);

    expect(wrapper.children().first()).toHaveClassName("spinner-md");
  });
});
