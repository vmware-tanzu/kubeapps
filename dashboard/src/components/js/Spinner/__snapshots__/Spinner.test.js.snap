// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner displays the basic spinner 1`] = `
<span
  className="spinner-container spinner-center"
>
  <span
    className="spinner spinner-center"
  >
    Loading...
  </span>
</span>
`;
